src/hotspot/share/gc/g1/g1CardTable.hpp

```diff
@@ -63,18 +61,24 @@ class G1CardTable : public CardTable {
     // The merge at the start of each evacuation round simply sets cards to dirty
     // that are clean; scanned cards are set to 0x1.
     //
-    // This means that the LSB determines what to do with the card during evacuation
-    // given the following possible values:
+    // This means that the LSB determines whether the card is clean or non-clean
+    // (LSB is 1 -> clean, LSB is 0 -> non-clean) given the following possible values:
     //
     // 11111111 - clean, do not scan
-    // 00000001 - already scanned, do not scan
+    // xxxxxxx1 - clean, already scanned, do not scan again (during GC only).
+    // 00000100 - dirty, needs to be scanned, dirty from remembered set (during GC only)
+    // 00000010 - dirty, needs to be scanned, contains reference to collection set.
     // 00000000 - dirty, needs to be scanned.
     //
-    g1_card_already_scanned = 0x1
+    g1_dirty_card = dirty_card,
+    g1_card_already_scanned = 0x1,
+    g1_to_cset_card = 0x2,
+    g1_from_remset_card = 0x4
```

Comment:
Could you outline the motivation for this more precise info? Is it for optimization or essentially for correctness?